### ISSUE TITLE
[Feature Request]: need version 1.21.1

### ISSUE THREAD

**Issue by Vitokhv** [terrarium-earth/Ad-Astra#637]
### Is your feature request related to a problem?

need version 1.21.1 for Fabric

### Mod Version

1.16.5

### Mod Loader Version

1.21.1

### Mod Loader

Fabric

**Comment by DRD2k8** [comment]
Same, I actually want it for 1.21 on both Fabric and NeoForge. The creator of this mod hasn't updated the mod in forever.

**Comment by YayBlaze** [comment]
bump

**Comment by LoneSimba** [comment]
bump

**Comment by NoahGirling** [comment]
bump

I would love to see a 1.21.1 port, especially when it's beginning to become one of the most recent hotspot for modding. Loads of mods have begun porting versions, why not this one too?

**Comment by DeanBeanBEER-WARE** [comment]
over a year and no answer on this, but active development on an outdated modloader version, is this a joke ?

**Comment by Radon8472** [comment]
> over a year and no answer on this, but active development on an outdated modloader version, is this a joke ?

I was reading, that there are still dependencies missing, that this mod needs before they can update

**Comment by WulfyGary** [comment]
bump

**Comment by evelant** [comment]
It seems this is abandoned and due to the unfortunate license.md (code is open, everything else is not) it can only be revived and updated if somebody re-does all of the art from scratch. It would be nice if projects that wanted an ARR license also put a clause such as "this converts to MIT license if this file is not updated by (insert date a year from now)".

**Comment by DeanBeanBEER-WARE** [comment]
> It seems this is abandoned and due to the unfortunate license.md (code is open, everything else is not) it can only be revived and updated if somebody re-does all of the art from scratch. It would be nice if projects that wanted an ARR license also put a clause such as "this converts to MIT license if this file is not updated by (insert date a year from now)".

Some dev's seem to be allergic to success, can't count anymore how many times i've seen projects like this just being abandoned out of pure laziness.

**Comment by CodexAdrian** [comment]
This project isn't abandoned, and we've already said we're working on a rewritten version (not in this repo) for 26.1, skipping 1.21.

**Comment by evelant** [comment]
That's unfortunate, a lot of mods and modpacks are blocked from moving from 1.20 to 1.21 by ad-astra. If it will never get a 1.21 version that will make transitioning those modpacks to modern versions a lot harder due to many other mods having 1.21 versions but not 26.1 yet. Would you consider accepting a PR here to update this to 1.21 so it's usable while the rewrite for 26.1+ is ongoing?

**Comment by CodexAdrian** [comment]
If someone feels like updating Ad Astra to 1.21, and making the changes to migrate from Botarium to Common Storage Lib, and then PRing that then maybe. But I don't foresee that happening without someone trying to sneak an AI pr by us. Depending on where we see the timeline for the launch of Ad Astra 2.0 we may end up doing it ourselves, but at the moment we feel our time is better spent making the rewrite

**Comment by evelant** [comment]
You would have a problem with a (thoroughly human reviewed) PR done with AI help? Frontier models are very good at the sort of "mechanical transformation" needed to switch out / upgrade APIs and could likely get it ported fairly quickly and accurately (I haven't looked deep enough at the code myself yet so that's an assumption on my part).

**Comment by CodexAdrian** [comment]
If I wanted an AI to update Ad Astra, I'd just ask Claude myself. Human written.

**Comment by BarelyHum4n** [comment]
> If someone feels like updating Ad Astra to 1.21, and making the changes to migrate from Botarium to Common Storage Lib, and then PRing that then maybe. But I don't foresee that happening without someone trying to sneak an AI pr by us. Depending on where we see the timeline for the launch of Ad Astra 2.0 we may end up doing it ourselves, but at the moment we feel our time is better spent making the rewrite

what a fucking lie
https://www.reddit.com/r/feedthebeast/comments/1spios9/comment/ojbzfat/

**Comment by CodexAdrian** [comment]
The port used Claude... and they posted it on CF, they didn't PR it. I'm not sure what part of it was a lie?

**Comment by BarelyHum4n** [comment]
> The port used Claude... and they posted it on CF, they didn't PR it. I'm not sure what part of it was a lie?

suddenly the issue is they used ai when they stated it was because of the texures

**Comment by CodexAdrian** [comment]
Yes, posting a port on CurseForge that uses our assets is a violation of our license. That has nothing to do with someone making a pull request using AI, which we wouldn't accept.

**Comment by evelant** [comment]
Hey, it's their project, they can license it how they want. Anyone can make their own 1.21 space mod if they want, even use the code from ad-astra since that part is freely licensed. Complaining about maintainers licensing choices and AI preferences doesn't do anyone any good. They put in a lot of work to make the mod, they can do what they want with it (even if I personally disagree with an ARR and "no ai" policy, I just won't license my own things that way).

**Comment by DeanBeanBEER-WARE** [comment]
just fork it yourself, and use it in your singleplayer world. Or make your own assets, if you want to use the underlying logic. That's what licensing is for. Im not agreeing on the AI policy either, modern development is composer based, but it's their software, so they can decide what happens to it, plain and simple.

**Comment by Radon8472** [comment]
> just fork it yourself, and use it in your singleplayer world. Or make your own assets, if you want to use the underlying logic. That's what licensing is for. Im not agreeing on the AI policy either, modern development is composer based, but it's their software, so they can decide what happens to it, plain and simple.

I do not fully agree with a "no AI" Policy, But I am Developer myself, and I understand this perspective.
In the Past there where several issues that where caused by AI Code that was not well reviewed by a human developer.

Even some examples where people used AI code to bring "virus-like" behaviors into minecraft mods.
And you should understand, that it is a very time consuming task for a developer to review a big piece of complex code, and make fully sure that this code is not dooing something mad or unintented.
